### DISCUSSION TITLE
Apple M1 build?

### DISCUSSION THREAD

**Discussion by JSCjr** [lichess-org/fishnet#150] [General]
Are there plans for a pre-compiled binary for the Apple M1 architecture anytime soon? I could have a go at building it for myself, but I'm not a Rust user so there's a fair bit of overhead involved.

**Comment by niklasf** [comment]
Yes, if you can test on the actual hardware that would be a great way to make progress on this.

The first step would be checking if the Stockfish build works. The build artifact on https://github.com/niklasf/Stockfish/actions/runs/400799007 includes `stockfish-macos-aarch64`. Could you please check if that's working? (Download https://tests.stockfishchess.org/api/nn/nn-62ef826d1a6d.nnue next to it, then something like `chmod +x stockfish-macos-aarch64 && ./stockfish-macos-aarch64 bench`).

**Comment by JSCjr** [comment]
> The first step would be checking if the Stockfish build works. The build artifact on https://github.com/niklasf/Stockfish/actions/runs/400799007 <https://github.com/niklasf/Stockfish/actions/runs/400799007> includes stockfish-macos-aarch64. Could you please check if that's working? (Download https://tests.stockfishchess.org/api/nn/nn-62ef826d1a6d.nnue <https://tests.stockfishchess.org/api/nn/nn-62ef826d1a6d.nnue> next to it, then something like chmod +x stockfish-macos-aarch64 && ./stockfish-macos-aarch64 bench).
> 


  Seems to run just fine - stdout attached.

**Reply by niklasf** [reply]
Thanks for testing. Attachments sent by email are not uploaded to GitHub, but if you say it didn't crash it should really be fine. I'll prepare a fishnet build with the new Stockfish binary included.

**Reply by JSCjr** [reply]
[test.txt](https://github.com/niklasf/fishnet/files/5744787/test.txt)

**Comment by niklasf** [comment]
Alright ... on to the full client. The latest development snapshot (~https://github.com/niklasf/fishnet/actions/runs/447564809~ https://github.com/niklasf/fishnet/actions/runs/447592719) now includes `fishnet-aarch64-apple-darwin`. Does it work for you, @JSCjr?

**Reply by niklasf** [reply]
Oops, fixed a copy & paste mistake: https://github.com/niklasf/fishnet/actions/runs/447592719

**Reply by JSCjr** [reply]
Looks like it's working! Rough eyeball estimate from the summary output is that it's about twice as fast as the x86 client.

**Reply by niklasf** [reply]
Yay! Published as v2.2.0.

**Reply by JSCjr** [reply]
Excellent - thanks for the quick turnaround.
